Src/IronPython/Runtime/Generator.cs

```diff
@@ -408,6 +409,10 @@ public sealed class PythonGenerator : IEnumerator, IEnumerator<object>, ICodeFor
 
             RestoreFunctionStack();
 
+            // The generator may be in the process of handling an exception
+            // and therefore the generator needs to save its state
+            RestoreCurrentException(_exceptionContext);
```

Comment:
This only covers `next()` coming from Python and does not cover those from `IEnumerable.MoveNext()`

Reply:
I think I did that in mine? https://github.com/slozier/ironpython3/commit/9f68deaf5cfb0b35b927c1d57adc1d682fbaff31

Reply:
Looks like it